portal-bridge/src/types/state/trie_walker.rs

```diff
@@ -0,0 +1,159 @@
+use alloy_primitives::{Bytes, B256};
+use anyhow::ensure;
+use eth_trie::{decode_node, node::Node};
+use hashbrown::HashMap as BrownHashMap;
+use serde::{Deserialize, Serialize};
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq, Eq)]
+pub struct AccountProof {
+    pub path: Vec<u8>,
+    pub proof: Vec<Bytes>,
+}
+
+pub struct ParentWithPath {
+    pub parent_hash: B256,
+    pub path_nibbles: Vec<u8>,
+}
+
+impl ParentWithPath {
+    pub fn new(parent_hash: B256, path_nibbles: Vec<u8>) -> Self {
+        Self {
+            parent_hash,
+            path_nibbles,
+        }
+    }
+}
+
+/// This struct takes in a root hash and a hashmap of changed nodes, then you can call an iterator
+/// which will return every proof to gossip
+pub struct TrieWalker {
```

Comment:
One thing crossed my mind right after submitting previous review.

I think it would make sense to change this structure to something like:

```rust

struct TrieWalkerNode {
  /// The encoded version of the trie node.
  encoded_node: Vec<u8>,
  /// The hash of the parent node. It is `None` only for root of the trie.
  parent: Option<B256>,
  /// Path from parent node to this node.
  parent_path: Vec<u8>,
}

pub struct TrieWalker {
  nodes: BrownHashMap<B256, TrieWalkerNode>,
}
```

This would remove the need for `ParentWithPath` and I think this would also remove ambiguity about naming.